lib/cpuinfo.c

```diff
@@ -49,10 +49,20 @@
 #include <sched.h>       /* sched affinity */
 #endif
 
+#include "pqos.h"
+
+#include "cpu_registers.h"
 #include "log.h"
 #include "cpuinfo.h"
 #include "types.h"
 #include "machine.h"
+#include "os_allocation.h"
+#include "allocation.h"
+
+/**
+ * This variable will hold the CPU vendor
+ */
+enum pqos_vendor cpu_vendor = PQOS_VENDOR_UNKNOWN;
```

Comment:
Vendor information could be added to pqos_cpuinfo structure. Then we won't need to add global variable.

Reply:
We can do that but this involves quite a bit of changes.  I need to pass the vendor flag to multiple functions. Also we need the vendor information even before the pqos_cpuinfo structure is initialized.  If you insist, I will do that. Let me know.

Reply:
The only function that uses vendor information before cpuinfo is initialized is init_functions. We can easily move vendor detection to cpuinfo_init function.

Reply:
There were few more functions we needed the vendor information like cpuinfo_init, cpuinfo_build_topo, detect_cpu and print_per_socket_config. Also I hade to re-arrange the code in rdtset/rdtset.c to make it work. Pushed 2 commits to take care of it.  Tested on both Intel and AMD. Please review. thanks